src/annotator/plugin/document.js

```diff
@@ -26,14 +35,16 @@ export default class DocumentMeta extends Plugin {
     this.events = {
       beforeAnnotationCreated: 'beforeAnnotationCreated',
     };
+
+    /** @type {Metadata} */
+    this.metadata = { title: document.title, link: [] };
+
+    this.baseURI = options.baseURI || baseURI;
+    this.document = options.document || document;
+    this.normalizeURI = options.normalizeURI || normalizeURI;
```

Comment:
I moved these assignments from `pluginInit` to here because TS only looks at the constructor of a class to infer what fields it has.

Reply:
good to know!